src/peers.py

```diff
@@ -0,0 +1,33 @@
+from ops.framework import Object, StoredState
+
+
+class KeepalivedPeers(Object):
+
+    state = StoredState()
+
+    def __init__(self, charm, relation_name):
+        super().__init__(charm, relation_name)
+        self._relation_name = relation_name
+        self._relation = self.framework.model.get_relation(self._relation_name)
+        self.framework.observe(charm.on.keepalived_initialized, self)
+
+    @property
+    def is_joined(self):
+        return self._relation is not None
```

Comment:
Interesting, peer relations seem to be None in install, but exist now in start (this may only be with 2.8 and the relation-created hook).